.github/workflows/rs-all-client/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+cmake_minimum_required( VERSION 3.15 )
+
+#
+# This tests whether we can link with realsense2-all and not have any missing external symbols.
+# Without realsense2-all, we'd need to link with:
+#     realsense2 realsense-file rsutils
+# Or, if DDS was enabled:
+#     realsense2 realsense-file rsutils fastcdr fastrtps foonathan_memory realdds
+# And the list gets longer if we add libraries.
+#
+# On Windows, no additional special libraries are needed.
+# On Linux, we have other dependencies:
+#     libusb udev
+# These are not part of realsense2-all, even though we depend on them!
+#
+
+project( rs-all-client )
+
+option( BUILD_SHARED_LIBS "Build using shared libraries" OFF )
+
+if( WIN32 )
+    # Take away the other configurations because they'd require we picked another link
+    # directory and target... keep it simple...
+    set( CMAKE_CONFIGURATION_TYPES "Release" )
```

Comment:
What if we wish to debug??

Reply:
NVM, test only is fine

Reply:
This is for the unit-test, which only runs in GHA and compiles a specific build.
If you need to debug (unlikely) just remove the line. You'd have to make sure everything else lines up.